### ISSUE TITLE
Is everything covered in tests?

### ISSUE THREAD

**Issue by fir3pho3nixx** [fir3pho3nixx/Windsor#19]
I don't believe it is you know. There are definitely some tests I would like to write around the assumptions  we have for some of the runtime observation's I have had while performance tuning this app. 

IServiceProvider creation thrashes memory badly. I want to create a test to guard against this.

**Comment by generik0** [comment]
@fir3pho3nixx I do have to say, after I did a test with the web app, it is crazy difficult to crosswire something after startup. You need both the container and service collection 
I think it’s support is very corner case.

My gut feeling is, if you ate worried about memory, we should not support it.
And just throw exception.

**Comment by fir3pho3nixx** [comment]
Yes, why I asked you to submit a PR purely dealing with the exception piece for ServiceProvider creation. This is a hot topic. Would like to see your name on this commit. Somehow it ended up becoming a public API exposed at the container level. Please I simply don’t understand why these are linked ... call me stupid if you like. I simply don’t understand the linkage.

**Comment by fir3pho3nixx** [comment]
We should have chatted tonight.
